[misc] perf: Use Object.keys for empty object check

sgorsh opened the pull request:
When using moment together with ChartJS, I found out that there is performance issue with `isObjectEmpty ` when rendering large datasets. This function takes around 30% of total time required to render the chart. 

When using `Object.keys`, it only takes ~2% of total time using Chrome 57. 

![screenshot 434](https://cloud.githubusercontent.com/assets/25081285/24832686/6ca33f82-1cb5-11e7-9b15-532000621e87.png)

![screenshot 431](https://cloud.githubusercontent.com/assets/25081285/24832688/74c23538-1cb5-11e7-8c8c-897f08e2eb94.png)

butterflyhug commented on src/lib/utils/is-object-empty.js:
```diff
@@ -1,8 +1,12 @@
+    } else {
+        var k;
+        for (k in obj) {
+            // even if its not own property I'd still call it non-empty
```
Object.keys only looks at own properties.  I don't personally think that behavior change is a deal breaker for this PR -- but I do think that we need to have consistent behavior in all browsers, so could you please do a `hasOwnProperty` check here?

sgorsh commented:
I've added `hasOwnProperty` check and also changed from `Object.keys` to `Object.getOwnPropertyNames` which I think is more correct in this case.

I've also created jsperf test ([https://jsperf.com/momentjs-isobjectempty-test/1](url)) and apparently `for..in` loop is slow only in Chrome. Firefox has almost the same speeds for all three methods.

![image](https://cloud.githubusercontent.com/assets/25081285/24837125/494a65ce-1d2d-11e7-8b0a-5e215bef8c2b.png)
![image](https://cloud.githubusercontent.com/assets/25081285/24837128/4cd550a0-1d2d-11e7-9d2a-4fccd537fc0f.png)

butterflyhug commented:
I just ran the jsperf test on my iPad, and the old for-in loop seems to be reliably faster in mobile Safari:

![img_0003](https://cloud.githubusercontent.com/assets/282994/24842923/8eb07872-1d6b-11e7-9e45-919216c270e5.PNG)

ichernev commented:
So chrome wins a lot, firefox wins a bit, iPad loses a bit? Chrome is the most popular browser so it makes sense to merge this.

ichernev commented:
Merged in d0db6976231663d5a8aca19a47cdd7bcaabeb630
